### PR TITLE
Adds Plastic Surgery; Fixes Disfigurement

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Implements TG's plastic surgery.

Plastic surgery replaces the previous surgery "face repair".

If the patient being operated on has a disfigured face, then you perform plastic surgery to reconstruct their face and remove the disfigurement.

However, if their face is fine, then, instead, it pops up a list of 10 random names (checks for species and gender) that allows the surgeon to give them a new identity.

If you're an abductor, you can name them whacky names.

Other changes:
- High cloneloss will now render you "unknown"
- Instead of getting head disfigurement from brute or burn damage, it now goes by their combined damages

Also fixes disfigurement, in general. `DISFIGURED` was utilized as a bitflag in several places, but it didn't properly work, as it wasn't plugged into anything. Anywhere that it was utilized, adding/removing it now works correctly

Code changes
- Cleans up how disfigurement is used; there's no need to update the head icon or regenerate the entire body's icons when it happens; they have no special icons for disfigurement
- anywhere that disfigurement of the head occurs, it now uses the proper proc.

:cl: Fox McCloud
add: Adds Plastic surgery; fix someone's face or give them a new identity!
tweak: Having more than 50 cloneloss will render you "Unknown"
tweak: head disfigurement requires you to have more than 50 combined brute and burn damage, rather than tracking it separately
fix: Fixes Cryoxadone and Rezadone not fixing disfigurement, Fixes fluorosulfuric acid not causing disfigurement at proper thresholds
/:cl:

### CODE REVIEW

File: code/modules/surgery/plastic_surgery.dm [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+				names += "Subject [target.gender == MALE ? "i" : "o"]-[pick("a", "b", "c", "d", "e")]-[rand(10000, 99999)]"
+			names += random_name(target.gender, species_names) //give one normal name in case they want to do regular plastic surgery
+		var/chosen_name = input(user, "Choose a new name to assign.", "Plastic Surgery") as null|anything in names
+		if(!chosen_name)
+			return
+		var/oldname = target.real_name
+		target.real_name = chosen_name
+		var/newname = target.real_name	//something about how the code handles names required that I use this instead of target.real_name
+		user.visible_message("[user] alters [oldname]'s appearance completely, they are now [newname]!", "<span class='notice'>You alter [oldname]'s appearance completely, they are now [newname].</span>")
+	if(ishuman(target))
```
Comment (Fox-McCloud) [inline code/modules/surgery/plastic_surgery.dm]: this is redundant.

**Comment by Aurorablade** [comment]
"If you're an abductor, you can name them whacky names." oh boy...

File: code/__DEFINES/combat.dm [hunk 2]
```diff
@@ -39,8 +39,7 @@
 #define IGNORESLOWDOWN	256
 #define GODMODE		4096
 #define FAKEDEATH	8192	//Replaces stuff like changeling.changeling_fakedeath
-#define DISFIGURED	16384	//I'll probably move this elsewhere if I ever get wround to writing a bitflag mob-damage system
```
Comment (Fox-McCloud) [inline code/__DEFINES/combat.dm]: every time we kill off a bitflag, I become *slightly* happier.

Comment (RyanSmake) [reply code/__DEFINES/combat.dm]: I agree so much.

File: code/modules/reagents/chemistry/reagents/toxins.dm [hunk 3]
```diff
@@ -611,11 +611,12 @@
 						return
 
 				if(!H.unacidable)
-					var/obj/item/organ/external/affecting = H.get_organ("head")
-					affecting.receive_damage(0, 75)
+					var/obj/item/organ/external/head/affecting = H.get_organ("head")
+					if(affecting)
+						affecting.receive_damage(0, 75)
+						affecting.disfigure()
```
Comment (Fox-McCloud) [inline code/modules/reagents/chemistry/reagents/toxins.dm]: The above damage is always enough to disfigure, but on the offhand chance that we eventually have some species that's hyper-resistant to burn damage.....

File: code/modules/surgery/plastic_surgery.dm [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+			for(var/_i in 1 to 9)
+				names += "Subject [target.gender == MALE ? "i" : "o"]-[pick("a", "b", "c", "d", "e")]-[rand(10000, 99999)]"
+			names += random_name(target.gender, species_names) //give one normal name in case they want to do regular plastic surgery
+		var/chosen_name = input(user, "Choose a new name to assign.", "Plastic Surgery") as null|anything in names
+		if(!chosen_name)
+			return
+		var/oldname = target.real_name
+		target.real_name = chosen_name
+		var/newname = target.real_name	//something about how the code handles names required that I use this instead of target.real_name
+		user.visible_message("[user] alters [oldname]'s appearance completely, they are now [newname]!", "<span class='notice'>You alter [oldname]'s appearance completely, they are now [newname].</span>")
```
Comment (Tayyyyyyy) [inline code/modules/surgery/plastic_surgery.dm]: The `they are`s should be replaced with `[target.p_they()] [target.p_are()]` here, I think

**Comment by Pennwick** [comment]
Any chance of making it so you could use an ID in the surgery to give the target that name? Or would this be unfeasible/too powerful?

**Comment by tigercat2000** [comment]
👍

**Pull request merged**